src/Cli/dotnet/commands/dotnet-workload/install/IWorkloadInstaller.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Workloads.Workload.Install
+{
+    internal interface IWorkloadInstaller
+    {
+        InstallationUnit GetInstallationUnit();
+
+        void InstallWorkloadManifest(string manifestId, string manifestVersion, string sdkFeatureBand);
```

Comment:
http://wiki.c2.com/?PrimitiveObsession
now C# have record. it is easier now. We should warp all these strings to a type, even if there is only one thing. If we worry about the perf, we can make them structs like packageid.

manifestId, manifestVersion, sdkFeatureBand, workloadId should all be separate types. Some of them may make sense to combine later when the need of abstraction is clear. Also, there should be only one way to present these IDs in our system. For example, manifestId can only be all lowercased. sdk feature band can only be a number or 5xx. We need to do validation, normalization when they come in to the system from domain boundary (parser etc...). I did the similar thing to packageid.